File: src/services/analytics.service.js
import { analyticsApi } from '../api/analytics.api';

export const analyticsService = {
  async payrollCostTrend(months = 6) {
    const res = await analyticsApi.getPayrollCostTrend(months);
    return res || [];
  },
  async payrollCostByDepartment() {
    const res = await analyticsApi.getPayrollCostByDepartment();
    return res || [];
  },
  async leaveUtilizationByType() {
    const res = await analyticsApi.getLeaveUtilizationByType();
    return res || [];
  },
  async leaveUtilizationByDepartment() {
    const res = await analyticsApi.getLeaveUtilizationByDepartment();
    return res || [];
  },
  async attendanceTrend(days = 14) {
    const res = await analyticsApi.getAttendanceTrend(days);
    return res || [];
  },
  async recruitmentFunnel() {
    const res = await analyticsApi.getRecruitmentFunnel();
    return res || [];
  },
};

File: src/main/java/com/hrms/hr_payroll_management_system/service/analytics/AnalyticsService.java
package com.hrms.hr_payroll_management_system.service.analytics;

import com.hrms.hr_payroll_management_system.dto.response.analytics.AnalyticsPointResponse;
import com.hrms.hr_payroll_management_system.dto.response.analytics.AttendanceTrendPointResponse;

import java.util.List;

public interface AnalyticsService {

    List<AnalyticsPointResponse> getPayrollCostTrend(int months);

    List<AnalyticsPointResponse> getPayrollCostByDepartment();

    List<AnalyticsPointResponse> getLeaveUtilizationByType();

    List<AnalyticsPointResponse> getLeaveUtilizationByDepartment();

    List<AttendanceTrendPointResponse> getAttendanceTrend(int days);

    List<AnalyticsPointResponse> getRecruitmentFunnel();
}